### PR TITLE
[Clang] Enable nsan instrumentation pass

### PR DESCRIPTION
**Pull request by alexander-shaposhnikov**
Enable nsan instrumentation pass

### CODE REVIEW

File: clang/lib/CodeGen/BackendUtil.cpp [hunk 1]
```diff
@@ -707,6 +708,9 @@ static void addSanitizers(const Triple &TargetTriple,
       MPM.addPass(createModuleToFunctionPassAdaptor(ThreadSanitizerPass()));
     }
 
+    if (LangOpts.Sanitize.has(SanitizerKind::NumericalStability))
+      MPM.addPass(NumericalStabilitySanitizerPass());
```
Comment (vitalybuka) [inline clang/lib/CodeGen/BackendUtil.cpp:712]: you may want late to introduce post-instrumentations as on line 693 of msan

**Review by vitalybuka — APPROVED** [review]

File: clang/test/CodeGen/nsan-basic.c [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-unknown -emit-llvm -o - -fsanitize=numerical %s | FileCheck %s
```
Comment (MaskRay) [inline clang/test/CodeGen/nsan-basic.c:1]: remove `-unknown-unknown`

**Review by MaskRay — APPROVED** [review]

File: clang/test/CodeGen/nsan-basic.c (hunk 2 above)
Comment (alexander-shaposhnikov) [reply clang/test/CodeGen/nsan-basic.c]: done

File: clang/lib/CodeGen/BackendUtil.cpp (hunk 1 above)
Comment (alexander-shaposhnikov) [reply clang/lib/CodeGen/BackendUtil.cpp:712]: debugging would be a bit harder - so would prefer to postpone adding them, but yeah, thanks for the suggestion, will keep it in mind.

**Pull request merged**